### DISCUSSION TITLE
where is the width and height information at?

### DISCUSSION THREAD

**Discussion by Intervel** [Chlumsky/msdf-atlas-gen#88] [General]
first of all, i want to thank you for making this great technique for free! love it so far!

thanks man!

anyway, im using json as an output : 

`{
    "atlas": {
        "type": "msdf",
        "distanceRange": 2,
        "size": 48,
        "width": 64,
        "height": 64,
        "yOrigin": "bottom"
    },
    "name": "raleway_regular",
    "metrics": {
        "emSize": 1,
        "lineHeight": 1.1739999999999999,
        "ascender": 0.94000000000000006,
        "descender": -0.23400000000000001,
        "underlineY": -0.10000000000000001,
        "underlineThickness": 0.050000000000000003
    },
    "glyphs": [
        {
            "unicode": 65,
            "advance": 0.67900000000000005,
            "planeBounds": {
                "left": -0.013666666666666617,
                "bottom": -0.030416666666666609,
                "right": 0.69466666666666677,
                "top": 0.74041666666666672
            },
            "atlasBounds": {
                "left": 0.5,
                "bottom": 26.5,
                "right": 34.5,
                "top": 63.5
            }
        },
        {
            "unicode": 66,
            "advance": 0.66800000000000004,
            "planeBounds": {
                "left": 0.064833333333333368,
                "bottom": -0.030416666666666609,
                "right": 0.64816666666666678,
                "top": 0.74041666666666672
            },
            "atlasBounds": {
                "left": 35.5,
                "bottom": 26.5,
                "right": 63.5,
                "top": 63.5
            }
        }
    ],
    "kerning": [
        {
            "unicode1": 66,
            "unicode2": 65,
            "advance": -0.014
        }
    ]
}`

i couldnt find the width and height information for each of Glyphs? 
also i tried to parse the information using freetype in my project.
im using 48 as pixel size. i got this results

![1](https://github.com/Chlumsky/msdf-atlas-gen/assets/86478025/ed4ddcb9-3a94-469c-99d2-b24781bcd22b)

why does it produce difference results? what am i doing wrong?
im using this command line

`msdf-atlas-gen.exe -font raleway_regular.ttf -fontname raleway_regular -type msdf -format png -size 48 -pxrange 2 -charset charset.txt -imageout test.png -json font.json -csv font.csv`

**Comment by Chlumsky** [comment]
Try subtracting left/right and bottom/top and you will get width and height.

**Reply by Intervel** [reply]
yo, thanks for the quick reply! i got width 34, and height 37. there is a 2 point gap there. 
is this come from "distanceRange": 2 ?

**Reply by Chlumsky** [reply]
There is no reason to expect the dimensions to be same as FreeType, there is extra border around glyphs (yes the distance range affects this), it will be rounded differently, etc.

**Comment by Intervel** [comment]
![2](https://github.com/Chlumsky/msdf-atlas-gen/assets/86478025/b07c347e-a707-4cbb-800b-7b24333eab84)

also, the how does this advance calculated?

![3](https://github.com/Chlumsky/msdf-atlas-gen/assets/86478025/59237975-1d69-4678-b9f2-adb59e3e9a6e)

![4](https://github.com/Chlumsky/msdf-atlas-gen/assets/86478025/76a23406-8990-42e6-b024-a3b3163facd3)

im using bitshift >> 6 to calculate the advance.x. i got this from this tutorial

https://learnopengl.com/In-Practice/Text-Rendering

the result produce 0.67900000000000005
